src/modules/frequency/application/frequency.service.query.interfaces.ts

```diff
@@ -1,6 +1,15 @@
-import { UserClassesDTO } from "./frequency.dtos";
+import {
+  EnrolledStudentDTO,
+  StudentClassAttendanceItemDTO,
+  StudentGeneralAttendanceResponseDTO,
+  UserClassesDTO } from "./dtos";
 
 export const FREQUENCY_QUERIES_TOKEN = "IFrequencyQueries";
-export interface IFrequencyQueries{
-    getMyClasses(userId: number): Promise<UserClassesDTO[]>;
-}
\ No newline at end of file
+export interface IFrequencyQueries {
+  getMyClasses(userId: number): Promise<UserClassesDTO[]>;
+  getGeneralAttendance(
+    date: Date,
+  ): Promise<StudentGeneralAttendanceResponseDTO[]>;
+  getStudentByClassAndDateAttendanceList(classId: number, date: Date): Promise<StudentClassAttendanceItemDTO[]>;
+  getStudentsByClassId(classId: number): Promise<EnrolledStudentDTO[]>;
```

Comment:
Is this a frequency method? Looks to me more of a students' (can stay for now, but I would consider moving it)
```suggestion
  getStudentsByClassId(classId: number): Promise<EnrolledStudentDTO[]>;
```

Reply:
Sim, provavelmente `frequency` não seja o melhor resource p/isso. Por uma questão de praticidade, acabei colocando nele.